test/fixtures/env-app/boot/test/bar.js

```diff
@@ -0,0 +1,5 @@
+process.bootFlags.push('barLoadedInTest');
+module.exports = function(app, callback) {
+  process.bootFlags.push('barStartedInTest');
+  callback();
+};
```

Comment:
AFAICT, your unit-test is using only `barLoadedInTest`, it does not check `barStartedInTest`. Please remove the unused exported function.